Split Unittest job for multiple python version

yunchu opened the pull request:
<!-- Contributing guide: https://github.com/openvinotoolkit/training_extensions/blob/develop/CONTRIBUTING.md -->

### Summary
Added metric of multiple python version to the unittest job in the pre-merge pipeline.
On this PR, 3.8, 3.9, and 3.10 are configured.

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test

<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated. -->

### Checklist

<!-- Put an 'x' in all the boxes that apply -->

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

yunchu committed:
Test unittest job for multiple python version

goodsong81 commented:
![image](https://user-images.githubusercontent.com/4629606/230616143-9575f6ce-6296-4c5b-aaf0-a710e8b493bc.png)
Might be missing?

yunchu commented:
> ![image](https://user-images.githubusercontent.com/4629606/230616143-9575f6ce-6296-4c5b-aaf0-a710e8b493bc.png) Might be missing?

It's related about the protection rule for the develop branch. i've renamed the unit tests in this PR but the rule tried to get the required unit test results with previous job name.
I've just updated the protection rule to omit old name of job and will add rename job as a required one once this PR is merged to develop.

goodsong81 reviewed (approved):
LGTM!
